src/apps/common/submodules/portListing/portListing.js

```diff
@@ -552,21 +554,21 @@ define(function(require) {
 					}));
 				});
 
-			template
+			textareaWrapper
 				.find('#add_comment')
 					.on('click', function(event) {
-						event.preventDefault();
-
-						var textarea = template.find('textarea[name="comment"]'),
-							comment = textarea.val();
+						self.portListingHelperPostComment(event, textarea, args, false);
+					});
 
-						self.portListingHelperAddComment($.extend(true, {}, args, {
-							data: {
-								comment: comment
-							}
-						}));
+			if (!monster.util.isSuperDuper()) {
+				this.return;
+			}
 
-						textarea.val('');
+			// Set event handlers for superduper options
+			textareaWrapper
+				.find('#add_private_comment')
+					.on('click', function(event) {
+						self.portListingHelperPostComment(event, textarea, args, true);
```

Comment:
Sending parameters like an object, this should looks like this:
```suggestion
						event.preventDefault();

						self.portListingHelperPostComment({
							textarea: textarea,
							args: args,
							isSuperDuperComment: true
						});
```